### ISSUE TITLE
Check wheter downloaded data are indeed of the size specified in RequestAsk

### ISSUE THREAD

**Issue by AuHau** [codex-storage/nim-codex#564]
The CID that is linked to a Request can have a different size compared to what is specified in the Request's Ask, we should double check that the downloaded size is not different.

**Comment by emizzle** [comment]
Agreed, but how can this be done without downloading the data? Is this information available in the manifest?

**Comment by emizzle** [comment]
Related to https://github.com/codex-storage/nim-codex/pull/1155
